prism/src/main/java/org/apache/falcon/service/EntitySLAMonitoringService.java

```diff
@@ -362,7 +367,7 @@ void addPendingEntityInstances(Date endTime) throws FalconException {
             if (EntityType.FEED.name().equals(entityType)
                     || isEntityRunning(EntityUtil.getEntity(entityType, entityName))) {
                 Date lastMonitoredInstanceTime = monitoredEntityBean.getLastMonitoredTime();
-                Date newCheckPointTime = endTime != null ? endTime : now();
+                Date newCheckPointTime = checkPointTime;
```

Comment:
So, checkPointTime is not expected to null. Better to add check and throw an exception.